### DISCUSSION TITLE
Bipes + HC-SR04 + ESP32 Eu não recebo sinal

### DISCUSSION THREAD

**Discussion by mekatronikautncba** [BIPES/BIPES#233] [Português]
Olá colegas.  Acabei de conhecer Bipes e fiquei agradavelmente surpreso.   Estou experimentando um esp32, mas não consigo imprimir a distância obtida de um sensor ultrassônico HC-SR04. Você tem um tópico no fórum para recomendar ou um exemplo para tentar?  muito obrigado

**Comment by jetannenbaum** [comment]
Hello,
   I can only test with the Raspberry Pi Pico and I get a distance response.  I'm assuming it works the same way for the ESP32.  Are you installing the library first and making sure it gets properly loaded?  Other than changing the defaults for the echo and trigger pins, the library should work as designed

**Reply by mekatronikautncba** [reply]
Hi Jim.  Thanks for your answer.  After testing for a while I saw a gray library block along with the ultrasonic sensor blocks.  Although I don't know exactly what it is like, I will verify that it is installed correctly. I changed the Trigger and Echo pins but the distance was always zero.  With Raspberry pico I had failed attempts with the simplest blocks, everything worked slower than real.  thank you so much

**Reply by mekatronikautncba** [reply]
Hello again.
I can observe that it works with a different library than the one you mentioned.
On the other hand when I install from the grey block the message appears that the library will be downloaded and installed, but I never see it reflected in the console.
It is time as we say in Argentina to "stop the ball and make another move". I do not want to take more time with this topic, I will try other sensors to learn more about the platform or try with raspberry pico.
Thank you very much my friend!

=== import time
=== from machine import Pin
=== 
=== from machine import time_pulse_us
=== 
=== #Based on: https://github.com/lemariva/uPySensors
=== class HCSR04:
===     def __init__(self, trigger_pin, echo_pin, echo_timeout_us=500*2*30):
===         self.echo_timeout_us = echo_timeout_us
===         self.trigger = Pin(trigger_pin, mode=Pin.OUT)
===         self.trigger.value(0)
===         self.echo = Pin(echo_pin, mode=Pin.IN)
=== 
===     def _send_pulse_and_wait(self):
===         self.trigger.value(0)
===         time.sleep_us(5)
===         self.trigger.value(1)
===         time.sleep_us(10)
===         self.trigger.value(0)
===         try:
===                     pulse_time = time_pulse_us(self.echo, 1, self.echo_timeout_us)
===                     return pulse_time
===         except OSError as ex:
===             if ex.args[0] == 110: # 110 = ETIMEDOUT
===                 raise OSError('Out of range')
===             raise ex
=== 
===     def distance_mm(self):
===         pulse_time = self._send_pulse_and_wait()
===         mm = pulse_time * 100 // 582
===         return mm
=== 
===     def distance_cm(self):
===         pulse_time = self._send_pulse_and_wait()
===         cms = (pulse_time / 2) / 29.1
===         return cms
=== 
=== 
=== ultraSoundSensor = HCSR04(trigger_pin=(22), echo_pin=(21), echo_timeout_us=10000)
=== while True:
===   print('d: ' + str(ultraSoundSensor.distance_mm()))
===   time.sleep(1)
=== 
=== 
=== 
d: 0
d: 0
d: 0
d: 0

**Comment by mekatronikautncba** [comment]
Hi Jim.  Thanks for your answer.  After testing for a while I saw a gray
library block along with the ultrasonic sensor blocks.  Although I don't
know exactly what it is like, I will verify that it is installed correctly.
I changed the Trigger and Echo pins but the distance was always zero.  With
Raspberry pico I had failed attempts with the simplest blocks, everything
worked slower than real.  thank you so much

**Comment by jetannenbaum** [comment]
Yes, you must install the library using the grey block while connected to the device.  Once complete, when you run the project, you should see the following in the console (I added a 1 second delay block so you could see the changing distance measurement)
![image](https://github.com/BIPES/BIPES/assets/865106/04d636c8-c5c5-447f-b4dd-5fe3b5c19376)

**Reply by mekatronikautncba** [reply]
Very good.  I will try it with esp 32 otherwise I will test with Raspberry pico again.  thank you so much

**Comment by joaodrj** [comment]
Já conseguiu resolver? se não eu posso ajuda-lo com isso.
